### PR TITLE
SONAR-8678 fixing multiple word search

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/SuggestionsAction.java [hunk 1]
```diff
@@ -112,10 +115,13 @@ SuggestionsWsResponse doHandle(String query) {
     try (DbSession dbSession = dbClient.openSession(false)) {
       return componentsPerQualifiers.stream().map(qualifier -> {
 
-        List<ComponentDto> componentDtos = dbClient.componentDao().selectByUuids(dbSession, qualifier.getComponentUuids());
+        List<String> uuids = qualifier.getComponentUuids();
+        List<ComponentDto> componentDtos = dbClient.componentDao().selectByUuids(dbSession, uuids);
+        List<ComponentDto> sortedComponentDtos = sortDtosByUuids(componentDtos, uuids);
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/SuggestionsAction.java]: Guava can help you here. Drop the explicit sorting thanks to:
```
Ordering.explicit(uuids).onResultOf(ComponentDto::uuid).immutableSortedCopy(componentDtos);
```

File: server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java [hunk 2]
```diff
@@ -28,6 +28,9 @@
   /** Maximum length of ngrams. */
   public static final int MAXIMUM_NGRAM_LENGTH = 15;
 
+  /** Pattern, that splits the user search input **/
+  public static final String SEARCH_TERM_TOKENIZER_PATTERN = "[\\:\\.\\s]+";
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java]: This constant is specific to context of ComponentIndexSearchFeature so I suggest to move it there.

File: server/sonar-server/src/test/java/org/sonar/server/component/ws/SuggestionsActionTest.java [hunk 3]
```diff
@@ -89,4 +93,25 @@ public void exact_match_in_one_qualifier() {
       .extracting(WsComponents.Component::getKey, WsComponents.Component::getOrganization)
       .containsExactly(tuple(dto.getKey(), organization.getKey()));
   }
+
+  @Test
+  public void sortByUuid() {
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/component/ws/SuggestionsActionTest.java]: can be dropped !

File: server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettingsElement.java [hunk 4]
```diff
@@ -25,17 +25,16 @@
 import java.util.SortedMap;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.settings.Settings.Builder;
+import org.sonar.server.component.index.ComponentIndexSearchFeature;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettingsElement.java]: MINOR Remove this unused import 'org.sonar.server.component.index.ComponentIndexSearchFeature'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**